Automate BZ 1240491 / CLI content-view puppet module tests

svtkachenko opened the pull request:
https://bugzilla.redhat.com/show_bug.cgi?id=1240491

```python
py.test tests/foreman/cli/test_contentview.py "-k ContentViewTestCase and test_positive_add_puppet_module"
============================= test session starts ==============================
platform linux2 -- Python 2.7.12, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
rootdir: /home/qui/code/robottelo, inifile: 
plugins: xdist-1.14, cov-2.3.1
collected 76 items

 74 tests deselected by '-k ContentViewTestCase and test_positive_add_puppet_module' 
================== 2 passed, 74 deselected in 147.48 seconds ===================
```

pondrejk commented on tests/foreman/cli/test_contentview.py:
```diff
@@ -1012,6 +1013,89 @@ def test_positive_add_custom_repo_by_name(self):
+                # Remove puppet module before next iteration
+                ContentView.puppet_module_remove({
+                    'content-view-id': content_view['id'],
+                    'uuid': puppet_module['uuid']
```
shouldn't we remove the same way we added (`by: puppet_module[by]`) so that we check that bot uuid and id work also for removing?

svtkachenko replied:
I thought about that but looks like `content-view puppet-module add` and `content-view puppet-module remove` use different ids. As it it possible to add puppet module by its global id, but not to remove. Or perhaps it is a bug. I'm not sure hot to check that as `content-view puppet-module remove`  does not display any ids except uuid.

```
hammer> puppet-module list
---|--------------------------|------------|---------|-------------------------------------
ID | NAME                     | AUTHOR     | VERSION | UUID                                
---|--------------------------|------------|---------|-------------------------------------
11 | ui_test_variables        | robottelo  | 0.2.0   | daa047a1-3e3a-425c-bb09-659474619ac3

hammer> content-view puppet-module add --content-view-id 66 --id 11
Puppet module added to content view

hammer> content-view puppet-module remove --content-view-id 66 --id 11
Couldn't remove puppet module from the content view:
  Error: content_view_puppet_module not found

hammer> content-view puppet-module list --content-view-id 66       
-------------------------------------|-------------------|-----------|--------
UUID                                 | NAME              | AUTHOR    | VERSION
-------------------------------------|-------------------|-----------|--------
daa047a1-3e3a-425c-bb09-659474619ac3 | ui_test_variables | robottelo | 0.2.0  
```

svtkachenko replied:
So I've checked API, indeed puppet modules associated with CV have its own ids. But I have no idea where I am supposed to fetch them using hammer.
```
GET https://sat6.com/katello/api/content_views/66/content_view_puppet_modules/
{
  "total": 2,
  "subtotal": 2,
  "page": 1,
  "per_page": 20,
  "error": null,
  "search": null,
  "sort": {
    "by": null,
    "order": null
  },
  "results": [
...
    {
      "id": 28,
      "uuid": "daa047a1-3e3a-425c-bb09-659474619ac3",
      "name": "ui_test_variables",
      "author": "robottelo",
      "created_at": "2017-02-24 15:17:51 +0200",
      "updated_at": "2017-02-24 15:17:51 +0200",
      "computed_version": "0.2.0"
    }
  ]
}
```

svtkachenko replied:
Looks like `content-view info` can help:
```
hammer> content-view info --id 66

Puppet Modules:         
 1) ID:     28
    UUID:   daa047a1-3e3a-425c-bb09-659474619ac3
    Name:   ui_test_variables
    Author: robottelo
```

pondrejk commented on tests/foreman/cli/test_contentview.py:
```diff
@@ -1012,6 +1013,89 @@ def test_positive_add_custom_repo_by_name(self):
+            'search': 'name={0} and version={1}'.format(
+                module['name'], module['version'])
+        })[0]
+        for by in ('uuid', 'id'):
```
maybe a more descriptive placeholder than `by` would be nice, up to you

svtkachenko commented:
I've done some refactoring in scope of this PR:
* Removed unneeded 'puppet-module remove` step from test that covers initial BZ. However, discussion about that step allowed to find another BZ.
* Added new tests for remove subcommand
* Removed 'puppet-module list' helper as such subcommand does not exist.

renzon commented on tests/foreman/cli/test_contentview.py:
```diff
@@ -1012,6 +1013,209 @@ def test_positive_add_custom_repo_by_name(self):
+            u'url': CUSTOM_PUPPET_REPO,
+        })
+        puppet_module = PuppetModule.list({
+            'search': 'name={0} and version={1}'.format(
```
Once you already have the dict you could use it:
```python
In [1]: module = {'name': 'versioned', 'version': '3.3.3'}

In [2]: "'search': 'name={name} and version={version}'".format(**module)
Out[2]: "'search': 'name=versioned and version=3.3.3'"
```

svtkachenko replied:
That's really cool hint, thank you! :) 👍

renzon commented on tests/foreman/cli/test_contentview.py:
```diff
@@ -1012,6 +1013,209 @@ def test_positive_add_custom_repo_by_name(self):
+        })
+        Repository.synchronize({'id': puppet_repository['id']})
+        puppet_module = PuppetModule.list({
+            'search': 'name={0} and version={1}'.format(
```
Same as previous comment
